tools/maint/update_settings_docs.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# Copyright 2021 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+"""Builds the emscripten website from source and creates a new commit & branch
+in the emscripten-site repository containing the changes."""
+
+import os
+import sys
+import subprocess
+
+script_dir = os.path.dirname(os.path.abspath(__file__))
+root_dir = os.path.dirname(os.path.dirname(script_dir))
+
+sys.path.append(root_dir)
+
+from tools.utils import path_from_root, read_file
+
+header = '''\
+.. _settings-reference:
+
+============================
+Emscripten Compiler Settings
+============================
+
+The following an complete list of settings that can be passed
+to emscripten via ``-s`` on the command line.  For example
+``-sASSERTIONS`` or ``-sASSERTIONS=0``.  For more details see the
+:ref:`emcc <emcc-s-option-value>` documentation.
+
+.. Auto-generated by update_settings_docs.py. **DO NOT EDIT**
+'''
+
+all_tags = {
+  'link': 'Only applicable during linking',
+  'compile+link': None,
+  'compile': 'Only applicable during compilation',
+  'experimental': 'This is an exeprimental setting',
+}
+
+output_file = path_from_root('site/source/docs/settings_reference.rst')
+
+
+def write_setting(f, setting_name, comment, tags):
+  # Convert markdown backticks to rst double backticks
+  prefixes = [
+    'emscripten_web_socket',
+    'emscripten',
+    'ENVIRONMENT_IS',
+    'stbi',
+    'emscripten_webgl_enable',
+    'library',
+  ]
+  for p in prefixes:
+    comment = comment.replace(p + '_*', p + '\\_*')
+  comment = comment.replace('`', '``')
+  comment = comment.replace('*', '\\*')
+  comment = comment.replace('"GL_"', '"GL\\_"')
```

Comment:
I guess GL can just go in the prefixes list above?

Reply:
I decided to go with modifying the existing comments rather than these magic replacements.

Reply:
That makes sense to me. What will happen when someone inevitably forgets to format a comment in settings.js the right way? Is there some part of the comment that we consider essential,  so we can check whether update_settings successfully parsed that?

Reply:
(also, do we still even need the double-backticks replacement?)

Reply:
If you mess up the syntax then the docs would fail to build

Reply:
Yes, all these replacements are gone now.